operators/swap_strips.py

```diff
@@ -1,33 +1,62 @@
 
 class SwapStrips(bpy.types.Operator):
     """
-    Swaps the 2 selected strips between them. More specific, places the first
-    strip in the channel and starting frame (frame_final_start) of the second
-    strip, and places the second strip in the channel and starting frame
-    (frame_final_end) of the first strip. If there is no space for the swap, it 
-    does nothing. If at least 1 of the selected strips is of effect type, the
-    operator won't have any effect.
+    Swaps the 2 strips between them. More specific, places the first strip in 
+    the channel and starting frame (frame_final_start) of the second strip, and 
+    places the second strip in the channel and starting frame (frame_final_end) 
+    of the first strip. If there is no space for the swap, it does nothing.
     """
     bl_idname = "power_sequencer.swap_strips"
     bl_label = "Swap Strips"
-    bl_description = "Swaps the 2 selected strips between them"
+    bl_description = "Swaps 2 strips between them"
     bl_options = {"REGISTER", "UNDO"}
+    
+    direction = bpy.props.EnumProperty(
+		name="Direction",
+		description="The direction to find the closest strip",
+		items=[("up", "Up", "The direction up from the selected strip"), \
+                ("down", "Down", "The direction down from the selected strip"), \
+                ("left", "Left", "The direction left from the selected strip"), \
+                ("right", "Right", "The direction right from the selected strip")],
+        default="up"
+	)
 
     @classmethod
     def poll(cls, context):
-        return len(bpy.context.selected_sequences) == 2
+        return len(bpy.context.selected_sequences) in range(1, 3)
 
     def execute(self, context):
         strip_1 = context.selected_sequences[0]
-        strip_2 = context.selected_sequences[1]
-        
+        if len(context.selected_sequences) == 1:
+            strip_2 = self.closest_direction_strip(context, \
+                    context.selected_sequences[0])
+            if None == strip_2:
+                return {'CANCELLED'}
+        else:
+            strip_2 = context.selected_sequences[1]
+            
         if strip_1.lock or strip_2.lock:
             return {'CANCELLED'}
         
         if hasattr(strip_1, 'input_1') or hasattr(strip_2, 'input_1'):
-            return {'CANCELLED'}
+            if strip_1.frame_final_start != strip_2.frame_final_start or \
+                    strip_1.frame_final_end != strip_2.frame_final_end:
+                return {'CANCELLED'}
+            
+            effect_strip = strip_1 if hasattr(strip_1, 'input_1') else strip_2
+            other_strip = strip_1 if effect_strip != strip_1 else strip_2
+            
+            effect_strip_channel = effect_strip.channel
+            other_strip_channel = other_strip.channel
+            
+            effect_strip.channel -= 1
+            other_strip.channel = effect_strip_channel
+            effect_strip.channel = other_strip_channel

```

Comment:
You can simplify the past 5 lines in Python with cross-assignment:
```py
other_strip.channel, effect_strip.channel  = effect_strip.channel, other_strip.channel
```

Reply:
As far as I know that translates to:
a = effect_strip.channel
b = other_strip.channel
other_strip.channel = a
effect_strip.channel = b

Unfortunately, that doesn't work, consider this case:
other_strips is a "normal" video strip. The above statement would have sent it to the effect strip channel, which is not empty, so other_strip lands with frame_final_start == effect_strip.frame_final_end.

Now, we can just change the order:
effect_strip.channel, other_strip.channel  = other_strip.channel, effect_strip.channel

Unfortunately again, that won't work. Effect strips can get "inside" other strips, so that would have worked only if all 32 channels are not empty. If there is at least 1 empty channel, effect_strip can be stored to, then setting the channel of the effect_strip to the channel of other_strip, would result in effect_strip landing in the empty channel, and not "inside" other_strip. After that, other_strip goes to its correct location, but we lost its previous location information, and effect_strip is not in a correct location.

We can solve that by storing the location of other_strip in a variable, and after the cross-assignment we can set effect_strips.channel to it, which leads to my code.